src/main/java/teammates/ui/webapi/GetCourseJoinStatusAction.java

```diff
@@ -42,12 +46,21 @@ private JsonResult getStudentJoinStatus(String regkey) {
         return getJoinStatusResult(student.isRegistered());
     }
 
-    private JsonResult getInstructorJoinStatus(String regkey) {
-        InstructorAttributes instructor = logic.getInstructorForRegistrationKey(regkey);
-        if (instructor == null) {
-            throw new EntityNotFoundException("No instructor with given registration key: " + regkey);
+    private JsonResult getInstructorJoinStatus(String regkey, boolean isCreatingAccount) {
+        if (isCreatingAccount) {
+            try {
+                AccountRequestAttributes accountRequest = logic.getAccountRequestForRegistrationKey(regkey);
+                return getJoinStatusResult(accountRequest.getRegisteredAt() != null);
+            } catch (EntityDoesNotExistException ednee) {
+                throw new EntityNotFoundException(ednee);
+            }
+        } else {
+            InstructorAttributes instructor = logic.getInstructorForRegistrationKey(regkey);
```

Comment:
If the previous `if` block always terminates, there is no need for `else`. This way we save up one level of indentation. This is known as the "no else return" pattern.